src/redux/selectors.js

```diff
@@ -62,7 +62,14 @@ export function getSource(state) {
 }
 
 export function getCode(state) {
-  return state.getIn(['REPL', 'code']);
+  if (state.getIn(['REPL', 'codeIndex']) === 0) {
```

Comment:
I would reuse `getCodeIndex()` here so if you change the shape of the state object, you only have to change one function.